### PR TITLE
feat: add new native connection panel

### PR DESCRIPTION
**Pull request by yichengchen**

### CODE REVIEW

File: ClashX/ViewControllers/Connections/ConnectionsViewController.swift [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+
+        leftWidthConstraint = leftTableView.widthAnchor.constraint(equalToConstant: 200)
+        leftTableView.makeConstraints {
+            [$0.leftAnchor.constraint(equalTo: view.leftAnchor),
+             $0.topAnchor.constraint(equalTo: view.topAnchor),
+             $0.bottomAnchor.constraint(equalTo: view.bottomAnchor),
+             leftWidthConstraint!]
+        }
+
+        (view as! ConnectionsViewControllerBaseView).leftWidthConstraint = leftWidthConstraint
```
Comment (miniLV) [inline ClashX/ViewControllers/Connections/ConnectionsViewController.swift:77]: view as`!` ConnectionsViewControllerBaseView, pls check if force unwrapping.

File: ClashX/ViewControllers/Connections/ViewModels/ConnectionDetailViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    @Published var destination = ""
+    @Published var applicationPath:String? = ""
+
+    var cancellable = Set<AnyCancellable>()
+
+    func accept(connection: ClashConnectionSnapShot.Connection?) {
+        cancellable.removeAll()
+        guard let connection else { return }
+        if let pid = connection.metadata.pid {
+            processName = "\(connection.metadata.processName ?? "Unknown") (\(pid))"
```
Comment (miniLV) [inline ClashX/ViewControllers/Connections/ViewModels/ConnectionDetailViewModel.swift:38]: "Unknown" need localization

File: ClashX/ViewControllers/Connections/ViewModels/ConnectionsViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+        let tableString:String = clash_getProggressInfo().toString().trimmingCharacters(in: .whitespacesAndNewlines)
+        if tableString.count < 0 { return [:] }
+        let processList = tableString.components(separatedBy: "\n")
+        var map = [String:String]()
+        for process in processList {
+            let infos = process.components(separatedBy: " ")
+            // fmt.Sprintf("%s %d %d\n", srcIP, srcPort, pid)
+            let srcIp = infos[0]
+            let srcPort = infos[1]
+            let pid = infos[2]
```
Comment (miniLV) [inline ClashX/ViewControllers/Connections/ViewModels/ConnectionsViewModel.swift:123]: Check if the index needs to be protected against out-of-bounds errors

example: infos.count = 1, get infos[2]

File: ClashX/ViewControllers/Connections/ViewModels/ConnectionsViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+                                         name: path.components(separatedBy: "/").last,
+                                         path: path)
+        applicationMap[pid] = info
+        return info
+    }
+}
+
+@available(macOS 10.15, *)
+class ConnectionsReq:WebSocketDelegate {
+    let socket = WebSocket(url: URL(string: ConfigManager.apiUrl.appending("/connections"))!)
```
Comment (miniLV) [inline ClashX/ViewControllers/Connections/ViewModels/ConnectionsViewModel.swift:169]: let socket = WebSocket(url: URL(string: ConfigManager.apiUrl.appending("/connections"))`!`)

Force unwrapping

File: ClashX/ViewControllers/Connections/Views/Cell/ConnectionLeftTextCellView.swift [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+             $0.leadingAnchor.constraint(equalTo: imageView.trailingAnchor, constant: 5),
+             $0.trailingAnchor.constraint(lessThanOrEqualTo: trailingAnchor)
+            ]
+        }
+        nameLabel.setContentCompressionResistancePriority(.defaultLow, for: .horizontal)
+        nameLabel.cell?.truncatesLastVisibleLine = true
+    }
+
+    func setup(with connection: ConnectionApplication) {
+        nameLabel.stringValue = connection.name ?? "Unknown"
```
Comment (miniLV) [inline ClashX/ViewControllers/Connections/Views/Cell/ConnectionLeftTextCellView.swift:47]: "Unknown" add localization

File: ClashX/ViewControllers/Connections/Views/Cell/ConnectionProxyClientCellView.swift [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+             $0.leadingAnchor.constraint(equalTo: imageView.trailingAnchor, constant: 4),
+             $0.trailingAnchor.constraint(lessThanOrEqualTo: trailingAnchor)
+            ]
+        }
+        nameLabel.setContentCompressionResistancePriority(.defaultLow, for: .horizontal)
+        nameLabel.cell?.truncatesLastVisibleLine = true
+    }
+
+    func setup(with connection: ClashConnectionSnapShot.Connection, type: ConnectionColume) {
+        nameLabel.stringValue = connection.metadata.processName ?? "Unknown"
```
Comment (miniLV) [inline ClashX/ViewControllers/Connections/Views/Cell/ConnectionProxyClientCellView.swift:47]: `Unknown` add localization

File: ClashX/ViewControllers/Connections/Views/ConnectionLeftPannelView.swift [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+                view = ConnectionApplicationCellView()
+            case .remote, .all, .hosts:
+                view = ConnectionLeftTextCellView()
+            }
+            view?.identifier = identifier
+        }
+
+        switch type {
+        case .local:
+            (view as! ConnectionApplicationCellView).setup(with: viewModel.localApplications[indexPath.item])
```
Comment (miniLV) [inline ClashX/ViewControllers/Connections/Views/ConnectionLeftPannelView.swift:173]: as`!` --> Force unwrapping

File: ClashX/ViewControllers/Connections/ConnectionsViewController.swift (hunk 1 above)
Comment (yichengchen) [reply ClashX/ViewControllers/Connections/ConnectionsViewController.swift:53]: this view is inited in loadView, no need to check again

**Pull request merged**